integrations/cypress/cypress/support/commands/index.ts

```diff
@@ -1,45 +1,52 @@
-import { makeGetClientFromSpec } from '../utils/spec';
-import { makeGiven } from './given';
-import { makeSDKCommands } from './add-sdk-commands';
+import type { CreateSimulation, Person } from '../types';
 import type { Auth0Result } from 'auth0-js';
-import './nextjs_auth0/get-user-info';
-import './nextjs_auth0/get-user-tokens';
-import { makeDestroySimulation } from './destroy-simulation';
+import { getConfig } from "../utils";
+import { registerGeneralCommands } from "./general";
+import { registerNextjsAuth0Commands } from "./nextjs_auth0";
+import { registerAuth0ReactCommands } from "./auth0_react";
 
 declare global {
-  namespace Cypress {
-    interface Chainable<Subject> {
-      createSimulation(options?: CreateSimulation): Chainable<Subject>;
-      destroySimulation(): Chainable<Subject>;
-      login(person?: Partial<Person>): Chainable<Token>;
-      logout(): Chainable<void>;
-      given(attrs?: Partial<Person>): Chainable<Person>;
-      out<S = unknown>(msg: string): Chainable<S>;
-      getUserInfo(accessToken: string): Chainable<Person>;
-      getUserTokens(person: Person): Chainable<Auth0Result & { scope: string }>;
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      task<T>(event: string, args?: any, options?: Partial<Loggable & Timeoutable>): Chainable<T>
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace Cypress {
+        interface Chainable<Subject> {
+            createSimulation(options?: CreateSimulation): Chainable<Subject>;
+
+            destroySimulation(): Chainable<Subject>;
+
+            login(person?: Partial<Person>): Chainable<string>;
+
+            logout(): Chainable<AUTWindow>;
+
+            given(attrs?: Partial<Person>): Chainable<Person>;
+
+            out<S = unknown>(msg: string): Chainable<S>;
+
+            getUserInfo(accessToken: string): Chainable<Person>;
+
+            getUserTokens(person: Person): Chainable<Auth0Result & { scope: string }>;
+
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            task<T>(event: string, args?: any, options?: Partial<Loggable & Timeoutable>): Chainable<T>
+        }
     }
-  }
 }
 
-const atom = createAtom<TestState>({});
-
-const ClientPort = Number(process.env.PORT || 4000);
+function setupAuth0CypressCommands() {
+    let config = getConfig();
 
-const getClientFromSpec = makeGetClientFromSpec({ atom, port: ClientPort });
+    let provider = config.sdk;
 
-Cypress.Commands.add('createSimulation', makeCreateSimulation({ atom, getClientFromSpec }));
+    registerGeneralCommands();
 
-Cypress.Commands.add('given', makeGiven({ atom, getClientFromSpec }));
+    if (provider === 'nextjs_auth0') {
+        registerNextjsAuth0Commands();
+    }
 
-Cypress.Commands.add('destroySimulation', makeDestroySimulation({ atom, getClientFromSpec }));
+    if (provider === 'auth0_react') {
+        registerAuth0ReactCommands();
+    }
```

Comment:
I think there should be an assertion of default values if there is no `sdk` value or there will be no login command registered and nothing will work.

Reply:
I've added a throw error here